### PR TITLE
PLANNER-1689 Implement ScoreDirectorFactory benchmark

### PR DESCRIPTION
**Pull request by triceo**

### CODE REVIEW

**Comment by triceo** [comment]
[Results of running this benchmark.](https://github.com/kiegroup/optaplanner/files/3751694/sdfbench1.zip)

**Comment by triceo** [comment]
[Summary spreadsheet to see the data more clearly.](https://docs.google.com/spreadsheets/d/1SP1QzVkGSiPah4WzLtlZD7Y3bk9FWcx5H0J3BY7-4ag/edit#gid=0)

File: optaplanner-examples/src/main/java/org/optaplanner/examples/app/ScoreDirectorFactoryBenchmarkApp.java [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+// TODO include investment
+// TODO include meetingscheduling
+// TODO include nurserostering
+// TODO include pas
+// TODO include projectjobscheduling
+// TODO include scrabble
+// TODO include tennis
+// TODO include travelingtournament
+// TODO include tsp
+public class ScoreDirectorFactoryBenchmarkApp {
```
Comment (ge0ffrey) [inline optaplanner-examples/src/main/java/org/optaplanner/examples/app/ScoreDirectorFactoryBenchmarkApp.java]: Normally, we do scoreDirector benchmarks simply by adding a bencharkConfig.xml and a line in the BenchmarkApp.

Is there a reason we can't cover these benchmarks by adding
optaplanner-examples/src/main/resources/org/optaplanner/examples/app/benchmark/generalOptaPlannerScoreDirectorBenchmarkConfigTemplate.xml.ftl and a line in GeneralOptaPlannerBenchmarkApp?

Comment (ge0ffrey) [reply optaplanner-examples/src/main/java/org/optaplanner/examples/app/ScoreDirectorFactoryBenchmarkApp.java]: Ah, I see, the easy score directors are different every time. Same for inc's etc.

Comment (ge0ffrey) [reply optaplanner-examples/src/main/java/org/optaplanner/examples/app/ScoreDirectorFactoryBenchmarkApp.java]: We might want to take this a step further, if we go down this route. See 
CloudBalancingSolveAllTurtleTest#overwritingEasyScoreCalculatorClass

If we have a register class for all score director variants, we can automate more testing and benchmarking.

Comment (triceo) [reply optaplanner-examples/src/main/java/org/optaplanner/examples/app/ScoreDirectorFactoryBenchmarkApp.java]: @ge0ffrey Not sure how this helps with benchmarking. I simply need something to run a benchmark of all SDF over all defined problems.

But perhaps I'm just not getting you. Would you mind explaining yourself a bit more?

**Pull request closed without merging**